Add extra guidance on mentored pair programming (#1133)
CodeYourFuture/curriculum: org-cyf-guides/content/pair-programming/mentored/index.md

+++
title = 'Mentored pair programming'
description = 'Practice writing code and getting support'
emoji = '👥'
weight = 2
+++

We assign **mentored pair programming** as a coursework assignment throughout the course.

All of the standard [pair programming guidance](../guidance) applies - make sure you're familiar with it.

**Before** a mentored pair programming session, learners should decide what they're going to pair on and share this with the volunteer, e.g. by sharing a link to a particular Codewars exercise, a particular piece of coursework, or some other problem.

**During** mentored pair programming, the **learner** should:

> * Explain their throught process

Reviewer: ```suggestion
* Explain their thought process
```

* Break down the problem
* Plan what code to write
* Write all of the code
* Check that it works

The **mentor** should:

> * **Ask questions** to get the trainee thinking. It's always better to ask than tell.

Reviewer: ```suggestion
* **Ask questions** to get the trainee thinking. It's always better to _ask_ than tell.
```

* Help the trainee think about breaking down the problem
* Support the trainee as they write code
* Help the trainee to debug their code

> * Stretch the trainee by bringing up extra concerns

Reviewer: ```suggestion
* Stretch the trainee by bringing up edge cases or complications
```

> This is different from other forms of pair programming, where both people may be expected to write code, or people may take turns at different roles. This is because the goal is different - the goal is teaching.

Reviewer: ```suggestion
This is different from other forms of pair programming, where both people may be expected to write code, or people may take turns at different roles. This is because the goal is different: the goal is **teaching**.
```
